main/pipeline/pipeline.cc

```diff
@@ -275,29 +272,27 @@ unique_ptr<parser::Node> runRBSRewrite(core::GlobalState &gs, core::FileRef file
     return node;
 }
 
-unique_ptr<parser::Node> runPrismParser(core::GlobalState &gs, core::FileRef file, const options::Printers &print) {
+parser::Prism::TranslateResult runPrismParser(core::GlobalState &gs, core::FileRef file,
+                                              const options::Printers &print) {
     Timer timeit(gs.tracer(), "runParser", {{"file", string(file.data(gs).path())}});
 
-    unique_ptr<parser::Node> nodes;
-    {
```

Comment:
Don't remove this `{ }` block. It limits the lifetime of the name table unfreezing to just these two lines. Without it, the name table will stay unfrozen until the end of this function call